react/src/views/Transfers/TransferShipmentsOverview/TransferShipmentsOverviewView.tsx

```diff
@@ -0,0 +1,171 @@
+import { useContext, useMemo } from "react";
+import { gql, useQuery } from "@apollo/client";
+import { Alert, AlertIcon, Button, Heading, Stack } from "@chakra-ui/react";
+import { Link } from "react-router-dom";
+import { GlobalPreferencesContext } from "providers/GlobalPreferencesProvider";
+import { SHIPMENT_FIELDS_FRAGMENT } from "queries/fragments";
+import { ShipmentsQuery } from "types/generated/graphql";
+import { AddIcon } from "@chakra-ui/icons";
+import { TableSkeleton } from "components/Skeletons";
+import { SelectColumnFilter } from "components/Table/Filter";
+import ShipmentsTable from "./components/ShipmentsTable";
+import { BaseOrgCell, BoxesCell, DirectionCell, StateCell } from "./components/TableCells";
+
+export const ALL_SHIPMENTS_QUERY = gql`
+  ${SHIPMENT_FIELDS_FRAGMENT}
+  query Shipments {
+    shipments {
+      ...ShipmentFields
+    }
+  }
+`;
+
+function TransferShipmentOverviewView() {
+  const { globalPreferences } = useContext(GlobalPreferencesContext);
+
+  // fetch shipments data
+  const { loading, error, data } = useQuery<ShipmentsQuery>(ALL_SHIPMENTS_QUERY);
+
+  // transform shipments data for UI
+  const graphqlToTableTransformer = (shipmentQueryResult: ShipmentsQuery | undefined) =>
+    shipmentQueryResult?.shipments.map((element) => {
+      if (globalPreferences?.availableBases && element?.sourceBase && element?.targetBase) {
+        const availableBaseIds = globalPreferences.availableBases.map((base) =>
+          parseInt(base.id, 10),
+        );
+        const sourceBaseId = parseInt(element.sourceBase.id, 10);
+        const targetBaseId = parseInt(element.targetBase.id, 10);
+
+        const shipmentRow = {
+          id: element.id,
+          direction: "To",
+          partnerBaseOrg: {
+            base: element.targetBase.name,
+            organisation: element.targetBase.organisation.name,
+          },
+          state: element.state,
+          boxes: 0,
+          lastUpdated: "",
+        };
+
+        // calculating direction
+        if (availableBaseIds.includes(sourceBaseId)) {
+          shipmentRow.direction = "To";
+          shipmentRow.partnerBaseOrg = {
+            base: element.targetBase.name,
+            organisation: element.targetBase.organisation.name,
+          };
+        } else if (availableBaseIds.includes(targetBaseId)) {
+          shipmentRow.direction = "From";
+          shipmentRow.partnerBaseOrg = {
+            base: element.sourceBase.name,
+            organisation: element.sourceBase.organisation.name,
+          };
+        }
+
+        // counting of boxes from details
+        const notDeletedDetails = element.details.filter((detail) => !detail.deletedOn);
```

Comment:
I was mistaken here, the BE only returns details with `deletedOn === null` : https://github.com/boxwise/boxtribute/blob/a5e626a35ae39c48a127386b3eed055bda23e1ad/back/boxtribute_server/business_logic/box_transfer/shipment/fields.py#L14

Reply:
Hence boxes will always be unique, too

Reply:
I removed the filter